src/lib/containers/dataaccess/SubmissionPage.tsx

```diff
@@ -30,8 +30,8 @@ import { FileHandleLink } from '../widgets/FileHandleLink'
 export type SubmissionPageProps = {
   /** The ID of the submission to view */
   submissionId: string | number
-  /** Invoked when a reviewer clicks "Reject". Note that the web request to reject the submission is not sent. We do this to continue using the existing rejection dialog written in SWC */
-  onRejectClicked: () => void
+  /** Invoked when a reviewer clicks "Reject". Provides a parameter for an external component to provide a reason for rejection, which will reject the submission when invoked. */
+  onRejectClicked: (onReject: (rejectedReason: string) => void) => void
```

Comment:
SWC gives passes a callback so this component can say when the reject button is clicked. That callback has a callback so the modal can confirm rejection and include the reason.

Is there a better way to do this than callback-ception?

Reply:
Most straightforward option is to re-implement the reason dialog in react, but I understand this workaround until we bite off that piece of work.